### PR TITLE
feat: Add signoff option

### PR DESCRIPTION
**Pull request by mbwhite**
Our current process required that the `git commit --signoff` is used to add a `signed-off-by` trailer to the commit message.

Adding an additional CLI option to permit this to be added. 

- Updated CLI
- Updated tests
- Updated readme.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "commit-and-tag-version",
-  "version": "12.0.0",
+  "version": "13.0.0",
```
Comment (TimothyJones) [inline package.json:3]: I think this version bump might have been included by accident?

File: test/core.spec.js [hunk 2]
```diff
@@ -1475,6 +1475,45 @@ describe('cli', function () {
       expect(gitArgs).toHaveLength(0);
     });
 
+
+    it('--signedoff adds signed-off-by to the commit message', async function () {
```
Comment (TimothyJones) [inline test/core.spec.js:1479]: It’s late here so maybe I’m missing something, but I can’t see how this test confirms what it says in the it statement- is it supposed to?

**Review by TimothyJones — COMMENTED** [review]
This is great, thanks so much for the PR! AND documentation and tests! 🙌🎉

I left a couple questions inline - Also, do you think the argument would be better named `—signoff`, to match the git argument that people might be looking for?

**Comment by TimothyJones** [comment]
`npm run format:fix` will fix the build error

File: package.json (hunk 1 above)
Comment (mbwhite) [reply package.json:3]: oops - I must have tested the tool in the wrong repo.

**Comment by mbwhite** [comment]
> This is great, thanks so much for the PR! AND documentation and tests! 🙌🎉
> 
> I left a couple questions inline - Also, do you think the argument would be better named `—signoff`, to match the git argument that people might be looking for?

thanks :-)

I'd debated with myself on that exact question.   Using `--signoff` to match git as you say is valid... but I settled on `--signedoff` as I was worried that it was very close to the tools `--sign` option.

Would `commit-and-tag-version --sign --signoff`  be confusing?  

Happy to go within either spelling :-)

File: test/core.spec.js (hunk 2 above)
Comment (mbwhite) [reply test/core.spec.js:1478]: I copied the previous test that was testing for the `--sign` option and how it affected the `git commit` command; 

Just to confirm - admit I've not dug into deep details of how the tests are structured - I've done the 'inverse' test and deliberately broke the code and the test flags this up...

```
  ● cli › with mocked git › --signedoff adds signed-off-by to the commit message

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

    @@ -1,8 +1,8 @@
      Array [
        "commit",
    -   "--signoff",
    +   "--wibble",
        "CHANGELOG.md",
        "package.json",
        "package-lock.json",
        "-m",
        "chore(release): 1.0.1",
  ```

Gives me some confidence that test would act as a regression check.

Comment (TimothyJones) [reply test/core.spec.js:1478]: Aha, yep, makes sense. I’m not the original author, so I’m not across exactly how all the tests work

**Comment by TimothyJones** [comment]
I see what you are saying about the ambiguity, but I reckon that’s a risk either way.

Even if the alternative is more ambiguous, I reckon that the consistency with git is worth it.

**Comment by mbwhite** [comment]
Thanks @TimothyJones

**Comment by TimothyJones** [comment]
Thank *you* for the contribution! Releasing 12.1.0 now, should be live shortly.

**Pull request merged**